gps/solver.go

```diff
@@ -105,12 +105,9 @@ type SolveParameters struct {
 	// typical case.
 	Downgrade bool
 
-	// Trace controls whether the solver will generate informative trace output
-	// as it moves through the solving process.
-	Trace bool
```

Comment:
I can put this back, but it's not doing anything, and I suspect it will confuse others in the future as it did me. (Maybe a separate issue if it's just clouding the water on this PR)

If it was set to `true`, then there was a validation check for a non-nil `TraceLogger` (even though nil checking the logger is the actual mechanism that triggers the trace logging internally).

If it was set to false, then it wasn't used. So you could pass a non-nil logger, and it would be passed along and tracing will take place, in spite of the `false` trace value.

So in it's current state it really amounts to "double check that `TraceLogger` is non-nil", but the only usages are setting a constant `true` value directly adjacent a call to set `TraceLogger` non-nil.